### DISCUSSION TITLE
Allow opening links in new tab and CSS styling in README.md

### DISCUSSION THREAD

**Discussion by aleksey-hoffman** [community/community#3782] [Other Feature Feedback, Questions, & Ideas]
Hey guys, can we discuss why markdown on Github is so limited and restrictive for no good reason? The security arguments don't make much sense.

Please consider the following:

1. Allow `target="_blank"` attribute for `<a>` tags.  Currently every link on README.md gets opened in the same tab. It doesn't improve security in any way (you can open links in a new tab with `CTRL + click`), it's just bad UX (e.g. a link to your Patreon page closes the Github project page).

2. Allow CSS styling in markdown. Github is not a place for simple libraries anymore. A lot of developers now publish their apps on Github and the README page is the front page of the app. It's practically impossible to create a beautiful front page without CSS.

**Proposed solution:**
Have 2 markdown parses on Github:
1. **Current parser**: default, restrictive, available for everyone;
2. **Unrestricted parser**: includes all CSS styling, available to all users by default, can be disabled for accounts that abuse CSS styling (e.g. after 5 abuse reports and a manual review by a Github employee)

**Comment by Andre601** [comment]
The Discussion Feedback category is more about GitHub's "Discussion" feature available for repositories.
You would probably be better of to make a [direct feedback](https://support.github.com/contact/feedback) to the GitHub Team in this case.



> 2. Allow CSS styling in markdown. Github is not a place for simple libraries anymore. A lot of developers now publish their apps on Github and the README page is the front page of the app. It's practically impossible to create a beautiful front page without CSS.

I see issues with this. Allowing even small CSS attribute changes could allow horrible things to happen. From flashy effects that cause epilepsy to just ugly looking text design (font, bg/font/link color, ...)
The README in particular is a Markdown file. And Markdown in itself doesn't allow custom CSS formatting. It's just something not worth to allow.

To the parse option and restricting stuff: This would only add more work to the GitHub Team and I believe they already have more than enough to do with the time they have at their disposal.

The only thing I do agree on is, that there should be a way to make links open in new tabs... Not just through the HTML-tag, but perhaps through a (GitHub flavourited) Markdown syntax. Maybe similar to how you can set a title, as in `[text](https://example.com 'Hover text')` but with like `_blank` to tell GitHub to add the `target="_blank"` attribute.

**Reply by aleksey-hoffman** [reply]
@Andre601 thanks for the suggestion. I will send a feedback request for opening links in a new tab.

> Allowing even small CSS attribute changes could allow horrible things to happen. 

That's indeed a problem, but as I mentioned, they could simply remove the ability to use custom CSS for the users who abuse it.

> This would only add more work to the GitHub Team

You have a good point, it would take a few weeks of work for the Github team to add this feature, even though the system is already implemented. Currently they just remove almost all HTML tags and all CSS styling from md files, which means they could simply add a few more tags and basic CSS rules to the "allow list" but even that would take a lot of testing and consideration. Yeah, I'm not sure if it's worth it.
